### PR TITLE
[webapp-perf] Migrate SettingItemMin and SettingItemMax to pure component, and replace anonymous function on props

### PR DESCRIPTION
**Pull request by saturninoabril**
#### Summary
Migrate SettingItemMin and SettingItemMax to pure component, and avoid/replace anonymous inline function on props

(Note: This is the first PR related to anonymous inline function on props)

#### Ticket Link
Jira ticket: [PLT-8296](https://mattermost.atlassian.net/browse/PLT-8296)

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed
- [x] Added or updated unit tests (required for all new features)

### CODE REVIEW

File: components/setting_item_max.jsx [hunk 1]
```diff
@@ -10,29 +10,117 @@ import SaveButton from 'components/save_button.jsx';
...
+
+        /**
+         * Array of inputs selection
+         */
+        inputs: PropTypes.array,
+
+        /**
+         * Client error
+         */
+        client_error: PropTypes.string,
```
Comment (jwilander) [inline components/setting_item_max.jsx]: Can we change this to be clientError?

Comment (saturninoabril) [reply components/setting_item_max.jsx]: Noted. In addition,
- Changed server_error to serverError
- Updated handleSubmit's parameter of EmailNotificationSetting

**Comment by jwilander** [comment]
This is quite a bit of code change this close to code complete, probably best to wait until on 4.7 unless anyone has a good reason we need it for 4.6

**Comment by saturninoabril** [comment]
Thanks! I'm good to have this to v4.7.

**Pull request merged**